test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -247,7 +247,115 @@ Describe "Invoke-WebRequest tests" -Tags "Feature" {
 
         $result = ExecuteWebCommand -command $command
         $result.Error.FullyQualifiedErrorId | Should Be "WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand"
+    }
+
+    It "Validate Invoke-WebRequest returns User-Agent where -NoProxy with -Proxy set" {
+
+        $command = "Invoke-WebRequest -Uri http://httpbin.org/headers -TimeoutSec 5 -NoProxy -Proxy http://localhost:8080"
+
+        $result = ExecuteWebCommand -command $command
+        ValidateResponse -response $result
+
+        # Validate response content
+        $jsonContent = $result.Output.Content | ConvertFrom-Json
+        $jsonContent.headers.'Accept-Encoding' | Should Match "gzip, ?deflate"
+        $jsonContent.headers.Host | Should Match "httpbin.org"
+        $jsonContent.headers.'User-Agent' | Should Match "WindowsPowerShell"
+    }
+
+    It "Validate Invoke-WebRequest error for -MaximumRedirection with Environment http_proxy set" {
+
+        # Store the current proxy.
+        $currentProxy = [System.Environment]::GetEnvironmentVariable("http_proxy")
+        [System.Environment]::SetEnvironmentVariable("http_proxy","http://localhost:8080")
 
+        $command = "Invoke-WebRequest -Uri 'http://httpbin.org/redirect/3' -MaximumRedirection 2 -TimeoutSec 5"
+
+        $result = ExecuteWebCommand -command $command
+        $result.Error.FullyQualifiedErrorId | Should Be "WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand"
```

Comment:
Not clear to me what this test is intended to validate.  Since `localhost:8080` is not a proxy, no redirection will actually happen and the error returned is that it can't contact the proxy.